pkg/comp-functions/functions/common/instance_namespace.go

```diff
@@ -0,0 +1,118 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vshn/appcat/v4/pkg/common/utils"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const roleBindingName = "appcat:services:read"
+const claimNsObserverSuffix = "-claim-ns-observer"
+
+func BootstrapInstanceNs(ctx context.Context, compositionName string, serviceName string, claimNs string, instanceNs string, svc *runtime.ServiceRuntime) error {
+	l := svc.Log
+
+	l.Info("creating namespace observer for mariadb claim namespace")
+	err := createNamespaceObserver(ctx, claimNs, compositionName, svc)
+	if err != nil {
+		return fmt.Errorf("cannot create namespace observer for claim namespace: %w", err)
+	}
+
+	l.Info("Creating namespace for " + serviceName + " instance")
+	err = createInstanceNamespace(ctx, serviceName, compositionName, claimNs, instanceNs, svc)
+	if err != nil {
+		return fmt.Errorf("cannot create %s namespace: %w", serviceName, err)
+	}
+
+	l.Info("Creating rbac rules for " + serviceName + " instance")
+	err = createNamespacePermissions(ctx, compositionName, svc)
+	if err != nil {
+		return fmt.Errorf("cannot create rbac rules for %s instance: %w", serviceName, err)
+	}
+
+	return nil
+}
+
+func getOrg(instance string, svc *runtime.ServiceRuntime) string {
+	ns := &corev1.Namespace{}
+
+	err := svc.GetObservedKubeObject(ns, instance+claimNsObserverSuffix)
+	if err != nil {
+		return ""
+	}
+	return ns.GetLabels()[utils.OrgLabelName]
+}
+
+func createNamespaceObserver(ctx context.Context, claimNs string, instance string, svc *runtime.ServiceRuntime) error {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: claimNs,
+		},
+	}
+
+	return svc.SetDesiredKubeObserveObject(ns, instance+claimNsObserverSuffix)
+}
+
+// Create the namespace for the mariadb instance
+func createInstanceNamespace(ctx context.Context, serviceName string, compName string, claimNamespace string, instanceNamespace string, svc *runtime.ServiceRuntime) error {
+
+	org := getOrg(compName, svc)
+	ns := &corev1.Namespace{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name: instanceNamespace,
+			Labels: map[string]string{
+				"appcat.vshn.io/servicename":     serviceName + "-standalone",
+				"appcat.vshn.io/claim-namespace": claimNamespace,
+				"appuio.io/no-rbac-creation":     "true",
+				"appuio.io/billing-name":         "appcat-" + serviceName,
+				"appuio.io/organization":         org,
+			},
+		},
+	}
+
+	return svc.SetDesiredKubeObjectWithName(ns, instanceNamespace, "ns-observer")
```

Comment:
```suggestion
	return svc.SetDesiredKubeObjectWithName(ns, instanceNamespace, "namespace-conditions")
```

The resource name for the instance-namespace is wrong. This will add the same object as another entry in the map. The old one is then considered gone and will be deleted.